lib/services/clusterconfig_test.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+type clusterConfig struct {
+}
+
+func (c clusterConfig) GetKind() string {
+	return ""
+}
+
+func (c clusterConfig) GetSubKind() string {
+	return ""
+}
```

Comment:
In this case I needed something that implemented `types.ClusterConfig` that wasn't a `types.ClusterConfigV3`. So I added `clusterConfig` and stubbed out the implementation of the `types.ClusterConfig` interface. I didn't actually implement a really usable mock here since I only needed a type assertion to fail.

There are a few test helper mocks/fakes floating around the code base that you can use. For other tests it might make more sense to create a local mock/fake of an interface that has just the specific bit you are interested in testing implemented.